ONSdigital/eq-survey-runner
# eq-364 Other option for radio/checkboxes

### What is the context of this PR?

[Trello card](https://trello.com/c/hNeyHf3v/642-5-respondent-radio-other-option-generates-a-free-text)

Adds an "other" option to radio buttons/checkboxes. This is defined in the schema as:

``` json
"options": [
    {
        "label": "Deep pan",
        "value": "deep pan"
    },
    {
        "label": "Thin",
        "value": "thin"
    },
    {
        "label": "Other",
        "value": "other",
        "other": {
          "label": "Please specify other"
        }
    }
],
```

This will generate the following pattern:

<img width="407" alt="screenshot 2016-10-11 08 38 18" src="https://cloud.githubusercontent.com/assets/930398/19262285/22de0968-8f8e-11e6-94a3-2388c4f78553.png">
### Additional changes
- Implemented new styling for radios/checkboxes with a minimum-width and a border around the selectable area.
- Re-factored `checkbox_group_widget.py` and `radio_group_widget.py` into a single `checkbox_radio_group_widget.py` as they were functionally the same.
### How to review
1. Start '0_checkbox_other' survey.
2. Confirm extra option works on both questions/pages.
3. Enter a value in "other".
4. Invoke a validation error on that page and check that the "other" value is retained.
5. Complete survey and check that "other" value is handled by the summary page.
